src/OrleansSQLUtils/CreateOrleansTables_MySql.sql

```diff
@@ -36,509 +27,415 @@ Implementation notes:
-	in	_faultZone INT,
-	in	_suspectingSilos TEXT,
-	in	_suspectingTimes TEXT,
 	in	_startTime DATETIME,
 	in	_iAmAliveTime DATETIME
 )
 BEGIN
 	DECLARE _ROWCOUNT INT;
 	START TRANSACTION;
-		
-		
-		-- There is no need to check the condition for inserting
-		-- as the necessary condition with regard to table membership
-		-- protocol is enforced as part of the primary key definition.
-		-- Inserting will fail if there is already a membership
-		-- row with the same
-		-- * DeploymentId 	= _deploymentId
-		-- * Address		= _address
-		-- * Port			= _port
-		-- * Generation		= _generation
-		--
-		-- For more information on table membership protocol insert see at
-		-- http://dotnet.github.io/orleans/Runtime-Implementation-Details/Runtime-Tables.html and at
-		-- https://github.com/dotnet/orleans/blob/master/src/Orleans/SystemTargetInterfaces/IMembershipTable.cs
-		INSERT INTO OrleansMembershipTable
-		(
-			DeploymentId,
-			Address,
-			Port,
-			Generation,
-			HostName,
-			Status,
-			ProxyPort,
-			RoleName,
-			InstanceName,
-			UpdateZone,
-			FaultZone,
-			SuspectingSilos,
-			SuspectingTimes,
-			StartTime,
-			IAmAliveTime
-		)
-		VALUES
-		(
-			_deploymentId,
-			_address,
-			_port,
-			_generation,
-			_hostName,
-			_status,
-			_proxyPort,
-			_roleName,
-			_instanceName,
-			_updateZone,
-			_faultZone,
-			_suspectingSilos,
-			_suspectingTimes,
-			_startTime,
-			_iAmAliveTime
-		);
-		
-		IF ROW_COUNT() = 1 
-		THEN
-			-- The transaction has not been rolled back. The following
-			-- update must succeed or else the whole transaction needs
-			-- to be rolled back.
-			UPDATE OrleansMembershipVersionTable
-			SET
-			 	Version		= _versionEtag + 1
-			WHERE
-				(DeploymentId	= _deploymentId AND _deploymentId IS NOT NULL)
-				AND (Version		= _versionEtag AND _versionEtag IS NOT NULL);
-		END IF;
-		-- Here the rowcount should always be either zero (no update)
-		-- or one (exactly one entry updated).
-        SET _ROWCOUNT = ROW_COUNT();
-		IF _ROWCOUNT = 1
-	 	THEN
-			COMMIT;
-	 	ELSE
-	 		ROLLBACK;
-	 	END IF;
-        SELECT _ROWCOUNT;
-	END$$
-
-	DELIMITER ;
+	INSERT INTO OrleansMembershipTable
+	(
+		DeploymentId,
+		Address,
+		Port,
+		Generation,
+		HostName,
+		Status,
+		ProxyPort,
+		StartTime,
+		IAmAliveTime
+	)
+	SELECT 
+		_deploymentId,
+		_address,
+		_port,
+		_generation,
+		_hostName,
+		_status,
+		_proxyPort,
+		_startTime,
+		_iAmAliveTime
+	WHERE NOT EXISTS
+	(
+	SELECT 1
+	FROM
+		OrleansMembershipTable
+	WHERE 
+		DeploymentId = _deploymentId AND _deploymentId IS NOT NULL
+		AND Address = _address AND _address IS NOT NULL
+		AND Port = _port AND _port IS NOT NULL
+		AND Generation = _generation AND _generation IS NOT NULL
+	);
+
+	UPDATE OrleansMembershipVersionTable
+	SET
+		Version = Version + 1
+	WHERE
+		DeploymentId = _deploymentId AND _deploymentId IS NOT NULL
+		AND Version = _version AND _version IS NOT NULL
+		AND ROW_COUNT() > 0;
+
+	SET _ROWCOUNT = ROW_COUNT();
+
+	IF _ROWCOUNT = 0
```

Comment:
This goes also to SQL Server, but I believe it would be more clear to explicitly `CAST` the integer here before returning it or commenting it will be interpreted as a `bool` in the calling code. There'll be dragons here in that SQL Server uses `BIT`, which is a different bit than that in MySQL, and MySQL equivalent might be `BOOL`, which is an alias to `TINYINT(1)` (if I remember correctly). Some more info: http://troels.arvin.dk/db/rdbms/#data_types-boolean.